harmony_netcdf_to_zarr/adapter.py

```diff
@@ -52,68 +56,72 @@ def __init__(self, message, catalog=None, config=None):
-            the input source defining the variables, if any, to subset from the item
-
-        Returns
-        -------
-        pystac.Item
-            a STAC item containing the Zarr output
         """
-        result = item.clone()
-        result.assets = {}
-
-        # Create a temporary dir for processing we may do
         workdir = mkdtemp()
         try:
-            # Get the data file
-            asset = next(v for k, v in item.assets.items() if 'data' in (v.roles or []))
-            input_filename = download(
-                asset.href,
-                workdir,
-                logger=self.logger,
-                access_token=self.message.accessToken,
-                cfg=self.config
-            )
-
-            name = generate_output_filename(asset.href, ext='.zarr')
-            root = self.message.stagingLocation + name
-
-            try:
-                store = self.s3.get_mapper(root=root, check=False, create=True)
-                netcdf_to_zarr(input_filename, store)
-            except Exception as e:
-                # Print the real error and convert to user-facing error that's more digestible
-                self.logger.error(e, exc_info=1)
-                filename = asset.href.split('?')[0].rstrip('/').split('/')[-1]
-                raise ZarrException(f'Could not convert file to Zarr: {filename}') from e
-
-            # Update the STAC record
-            result.assets['data'] = Asset(root, title=name, media_type='application/x-zarr', roles=['data'])
-
-            # Return the STAC record
-            return result
+            items = list(self.catalog.get_items())
+            netcdf_urls = get_netcdf_urls(items)
+
+            local_file_paths = download_granules(netcdf_urls, workdir,
+                                                 self.message.accessToken,
+                                                 self.config, self.logger)
+
+            # Enable during DAS-1379:
+            # dimensions_mapping = DimensionsMapping(local_file_paths)
+
+            if len(local_file_paths) == 1:
+                output_name = generate_output_filename(netcdf_urls[0],
+                                                       ext='.zarr')
+            else:
+                # Mimicking PO.DAAC Concise: for many-to-one the file name is
+                # "<collection>_merged.zarr".
+                collection = self._get_item_source(items[0]).collection
+                output_name = f'{collection}_merged.zarr'
+
+            zarr_root = path_join(self.message.stagingLocation, output_name)
+            zarr_store = self.s3.get_mapper(root=zarr_root, check=False,
+                                            create=True)
+
+            if len(local_file_paths) == 1:
+                # Temporarily retain old version of NetCDF-to-Zarr service
+                netcdf_to_zarr(local_file_paths[0], zarr_store)
+            else:
+                raise NotImplementedError('Concatenation is not yet supported.')
```

Comment:
I'm confused - isn't this where we initiate our aggregation processing?  Above code addresses many-to-one options, why raise an exception here?

Reply:
Yeah exactly. My aim here was to ensure that we could deploy this and continue service for a single granule input, but we've not yet actually done the bit to write many-to-one input (that's DAS-1379).

Reply:
I've added a short-lived TODO comment referring to DAS-1379.